tools/generator/rules.py

```diff
@@ -21,14 +25,16 @@ def make_rules(game, out_dir: Path) -> None:
         'game_state.hh'
     ]
     for tpl in base_files:
-        gen.template(tpl)
+        gen.template(tpl, out_name=os.path.join('src', tpl))
 
     for action in game["function"]:
         if not action.get("fct_action"):
             continue
-        gen.template('action_template.hh',
-                     out_name='action_{}.hh'.format(action['fct_name']),
-                     action=action)
-        gen.template('action_template.cc',
-                     out_name='action_{}.cc'.format(action['fct_name']),
-                     action=action)
+        for ext in ('hh', 'cc'):
+            gen.template('action_template.' + ext,
+                         out_name=os.path.join(
+                             'src',
+                             'action_{}.{}'.format(action['fct_name'], ext)),
```

Comment:
I think you can just use `src/action_{}.{}` here instead of the `join()`.

Reply:
ok